app/javascript/components/StaffDashboard/index.js

```diff
@@ -35,17 +36,18 @@ class StaffDashboard extends React.Component {
       });
       return;
     }
-    this.setState(prevState => ({
-      participants: prevState.trie.get(searchVal),
-    }));
+    const participants = this.state.trie.get(searchVal);
+    this.setState({
+      participants,
+    });
   }
 
   render() {
-    let participantsList = this.state.participants.map(p => (
-      <ParticipantCard key={p.id} participant={p}></ParticipantCard>
+    let participantsList = this.state.participants.map((p, i) => (
+      <ParticipantCard key={i} participant={p}></ParticipantCard>
```

Comment:
Using index as key is bad. You should use p.id here or something that doesn't change. You can read more about react keys/their purpose [here](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318)